server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ExecuteService.kt

```diff
@@ -77,20 +79,34 @@ class ExecuteService(
 
     fun test(cancelChecker: CancelChecker, params: TestParams): TestResult {
         val targets = selectTargets(cancelChecker, params.targets)
-        var result = build(cancelChecker, targets, params.originId)
-        if (result.isNotSuccess) {
-            return TestResult(result.statusCode)
-        }
         val targetsSpec = TargetsSpec(targets, emptyList())
 
+        var bazelTestParamsData: BazelTestParamsData? = null
+        try {
+            if (params.dataKind == BazelTestParamsData.DATA_KIND) {
+                val rawParams = params.data.toString()
+                bazelTestParamsData = Gson().fromJson(rawParams, BazelTestParamsData::class.java)
+            }
+        } catch (e: Exception) {
+            bspClientLogger.warn("Failed to parse BazelTestParamsData: $e")
+        }
+
+        var baseCommand = when (bazelTestParamsData?.coverage) {
+            true -> bazelRunner.commandBuilder().coverage()
+            else -> bazelRunner.commandBuilder().test()
+        }
+
         // TODO: handle multiple targets
-        withBepServer(params.originId, params.targets.single()) { bepReader ->
-            result = bazelRunner.commandBuilder().test()
-                .withTargets(targetsSpec)
-                .withArguments(params.arguments)
-                .withFlag(BazelFlag.color(true))
-                .executeBazelBesCommand(params.originId, bepReader.eventFile.toPath())
-                .waitAndGetResult(cancelChecker, true)
+        val result = withBepServer(params.originId, params.targets.single()) { bepReader ->
+            run {
+                baseCommand
+                    .withTargets(targetsSpec)
+                    .withArguments(params.arguments)
+                    .withFlag(BazelFlag.buildEventBinaryPathConversion(false))
```

Comment:
why is that needed?

Reply:
It provides file:// instead of bytestream:// paths in the outputs when using remote cache. ([--build_event_binary_file_path_conversion](https://bazel.build/reference/command-line-reference#flag--build_event_binary_file_path_conversion)).

Reply:
Could you put that explanation as a comment in the code?

Reply:
Added